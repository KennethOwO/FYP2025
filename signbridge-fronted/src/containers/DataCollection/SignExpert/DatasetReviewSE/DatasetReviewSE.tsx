import React from "react";
import DatasetReview from "../../components/DatasetReview/DatasetReview";

const DatasetReviewSE = () => {
  return <DatasetReview user="signexpert" />;
};

export default DatasetReviewSE;
