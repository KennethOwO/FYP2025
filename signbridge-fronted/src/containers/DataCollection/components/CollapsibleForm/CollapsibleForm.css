.collapsible-container {
    margin-bottom: 10px;
    width: 85%;
}

.collapsible-content {
    position: relative;
    top: 50%;
    left: 50%;
    transform: translate(-50%, 0);
    width: 100%;
    max-width: 100%;
    margin-top: 10px;
    margin-bottom: 20px;
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
    border-radius: 8px;
    transition: width 0.2s;
    background-color: white;
}

.collapsible-content-header {
    cursor: pointer;
    display: flex;
    justify-content: flex-start;
    align-items: center;
    padding: 20px;
    border-radius: 8px 8px 0 0;
}

.collapsible-content.not-opened {
    width: 85%;
    cursor: pointer;
}

.collapsible-content.not-opened:hover {
    width: 100%;
    background-color: rgb(224, 224, 224);
}

.collapsible-content.opened {
    width: 100%;
    border: 1px solid #a3a3a3;
}

.collapsible-content-header.admin {
    background-color: #a4825e;
}

.collapsible-content-header.signexpert {
    background-color: rgb(198, 198, 198);
}

.collapsible-content-background {
    padding: 20px;
}

.collapsible-content-details {
    height: auto;
    justify-content: space-evenly;
    margin-bottom: 30px;
}

.collapsible-content-header .number {
    width: 20%;
}

.collapsible-content-header .dateTime {
    width: 35%;
}

.collapsible-content-header .status {
    width: 35%;
}

.collapsible-content-header .expand-icon {
    margin-left: auto;
}

.personal-details,
.sentence-details,
.content-left {
    flex: 1;
    margin: 0 10px;
}

.personal-details h2,
.sentence-details h2,
.content-left h2 {
    display: inline-block;
    color: #3a3a3a;
    font-size: 22px;
    margin-bottom: 15px;
}

.personal-details .ant-descriptions {
    margin-bottom: 20px;
    padding: 10px;
}

.personal-details .ant-descriptions-bordered {
    border: 1px solid #ccc;
    border-radius: 4px;
}

.personal-details .ant-descriptions-item-label {
    font-size: 20px;
    color: rgb(68, 68, 68) !important;
    /* box-shadow: 0px 0px 4px rgba(0, 0, 0, 0.2); */
    background-color: #ececec !important;
    width: 25%;
    font-weight: 450;
}

/* .personal-details.admin .ant-descriptions-item-label {
    background-color: #FFF1C5 !important;
}

.personal-details.signexpert .ant-descriptions-item-label {
    background-color: #E7F0FD !important;
} */

.personal-details .ant-descriptions-item-content {
    font-size: 18px;
    display: flex;
    justify-content: center;
    align-items: center;
}

.sentence-details-content p {
    border-radius: 5px;
    padding: 10px;
    background-color: #ececec;
    font-size: 18px;
    margin: 0 10px;
}

.collapsible-content-footer {
    display: flex;
    justify-content: space-between;
    align-items: flex-end;
    flex-wrap: wrap;
    width: 100%;
}

.collapsible-content-footer .content-left {
    flex-grow: 1;
    margin-top: -10px;
}

.collapsible-content-footer .content-left .ant-descriptions-bordered {
    border: 1px solid #ccc;
    border-radius: 4px;
}

.ant-descriptions.ant-descriptions-bordered .ant-descriptions-row {
    border-bottom: 1px solid #e8e8e8;
}

.collapsible-content-footer .content-left .ant-descriptions {
    max-width: 100%;
    padding: 10px;
}

.collapsible-content-footer .content-left .ant-descriptions-item-label {
    font-size: 20px;
    color: rgb(68, 68, 68) !important;
    background-color: #ececec !important;
    width: 25%;
    font-weight: 450;
}

.collapsible-content-footer .content-left .ant-descriptions-item-content {
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 16px;
    color: red;
    width: 100%;
    box-sizing: border-box;
    text-align: center;
}

.collapsible-content-footer .buttons-right {
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-end;
    margin-top: 10px;
}

.dataset-button-container {
    display: flex;
    /* flex-wrap: wrap; */
    justify-content: flex-end;
}

.dataset-button-individual {
    margin-left: 10px;
}

@media (max-width: 480px) {
    .collapsible-container {
        width: 100%;
    }

    .collapsible-content-header {
        flex-direction: column;
        align-items: flex-start;
    }

    .collapsible-content-header .number,
    .collapsible-content-header .status,
    .collapsible-content-header .dateTime {
        width: 100%;
        margin-bottom: 5px;
        font-size: 16px;
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
    }

    .buttons-right {
        align-items: center;
        justify-items: center;
    }

    .collapsible-content-footer .content-left {
        width: 100%;
        display: flex;
        flex-direction: column;
        margin: 0;
        padding: 0;
        text-align: left;
    }

    .collapsible-content-footer .buttons-right {
        width: 100%;
        justify-content: center;
        margin-top: 10px;
    }

    .dataset-button-container {
        display: flex;
        justify-content: center;
        gap: 10px;
    }

    .dataset-button-individual {
        width: 100%;
        margin-bottom: 10px;
    }

    .personal-details h2,
    .sentence-details h2,
    .content-left h2 {
        font-size: 18px;
    }

    .sentence-details-content p {
        font-size: 14px;
        padding: 8px;
    }

    .personal-details .ant-descriptions-item-content,
    .collapsible-content-footer .content-left .ant-descriptions-item-content {
        font-size: 12px;
    }

    .personal-details .ant-descriptions-item-label,
    .collapsible-content-footer .content-left .ant-descriptions-item-label {
        font-size: 14px;
    }

    .expand-icon {
        position: absolute;
        right: 20px;
        top: 20px;
        margin-top: 0;
    }
}

/* Devices between 481px and 640px (mobile landscape) */
@media (min-width: 481px) and (max-width: 640px) {
    .collapsible-container {
        width: 90%;
    }

    .collapsible-content-header {
        padding: 15px;
        flex-direction: column;
        align-items: flex-start;
    }

    .personal-details h2,
    .sentence-details h2,
    .content-left h2 {
        font-size: 20px;
    }

    .collapsible-content-header .number,
    .collapsible-content-header .status,
    .collapsible-content-header .dateTime {
        font-size: 16px;
        margin-left: 10px;
    }

    .collapsible-content-header .number {
        width: 100%;
        font-size: 16px;
    }

    .collapsible-content-header .status {
        width: 100%;
        font-size: 16px;
    }

    .collapsible-content-header .dateTime {
        width: 100%;
        font-size: 16px;
    }

    .collapsible-content-header .number,
    .collapsible-content-header .status,
    .collapsible-content-header .dateTime {
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
    }

    .collapsible-content-footer {
        flex-direction: column;
        /* Stack elements vertically */
        align-items: center;
        /* Center elements horizontally */
    }

    .collapsible-content-footer .content-left {
        width: 100%;
        display: flex;
        flex-direction: column;
        margin: 0;
        padding: 0;
        text-align: left;
    }

    .collapsible-content-footer .buttons-right {
        width: 100%;
        justify-content: center;
        margin-top: 10px;
    }

    .dataset-button-container {
        display: flex;
        justify-content: center;
        gap: 10px;
    }

    .dataset-button-individual {
        margin-left: 10px;
    }

    .sentence-details-content p {
        font-size: 16px;
        padding: 8px;
    }

    .personal-details .ant-descriptions-item-content,
    .collapsible-content-footer .content-left .ant-descriptions-item-content {
        font-size: 14px;
    }

    .personal-details .ant-descriptions-item-label,
    .collapsible-content-footer .content-left .ant-descriptions-item-label {
        font-size: 16px;
    }

    .expand-icon {
        position: absolute;
        right: 20px;
        top: 20px;
        margin-top: 0;
    }
}

/* Devices between 641px and 768px (tablets and small devices) */
@media (min-width: 641px) and (max-width: 768px) {
    .collapsible-container {
        width: 85%;
    }

    .collapsible-content-header {
        padding: 18px;
    }

    .personal-details h2,
    .sentence-details h2,
    .content-left h2 {
        font-size: 20px;
    }

    .sentence-details-content p {
        font-size: 18px;
    }

    .collapsible-content-footer {
        flex-direction: column;
    }

    .collapsible-content-footer .content-left {
        width: 100%;
        display: flex;
        flex-direction: column;
        margin: 0;
        padding: 0;
        text-align: left;
    }

    .collapsible-content-footer .buttons-right {
        width: 100%;
        justify-content: center;
        margin-top: 10px;
    }

    .dataset-button-container {
        display: flex;
        justify-content: center;
        gap: 10px;
    }

    .dataset-button-individual {
        margin-left: 10px;
    }

    .collapsible-content-header .number,
    .collapsible-content-header .status,
    .collapsible-content-header .dateTime {
        font-size: 17px;
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
    }

    .personal-details .ant-descriptions-item-content,
    .collapsible-content-footer .content-left .ant-descriptions-item-content {
        font-size: 14px;
    }

    .personal-details .ant-descriptions-item-label,
    .collapsible-content-footer .content-left .ant-descriptions-item-label {
        font-size: 16px;
    }

    .expand-icon {
        position: absolute;
        right: 20px;
        top: 20px;
        margin-top: 0;
    }
}

/* For devices with a width between 769px and 1024px (tablets) */
@media (min-width: 769px) and (max-width: 1024px) {
    .collapsible-container {
        width: 85%;
    }

    .collapsible-content-header {
        flex-direction: row;
        padding: 20px;
    }

    .collapsible-content-footer {
        flex-direction: column;
    }

    .collapsible-content-footer .content-left {
        width: 100%;
        display: flex;
        flex-direction: column;
        margin: 0;
        padding: 0;
        text-align: left;
    }

    .collapsible-content-footer .buttons-right {
        width: 100%;
        justify-content: center;
        margin-top: 10px;
    }

    .dataset-button-container {
        display: flex;
        justify-content: center;
        gap: 10px;
    }

    .dataset-button-individual {
        margin-left: 10px;
    }

    .collapsible-content-header .number,
    .collapsible-content-header .status,
    .collapsible-content-header .dateTime {
        font-size: 20px;
        white-space: nowrap;
        /* Prevents text from wrapping */
        overflow: hidden;
        /* Hides any overflow text */
        text-overflow: ellipsis;
        /* Adds ellipsis when text overflows */
    }

    .expand-icon {
        position: absolute;
        right: 20px;
        top: 20px;
        margin-top: 0;
    }
}

@media (min-width: 1025px) {
    .collapsible-content.not-opened {
        width: 90%;
        cursor: pointer;
    }

    .collapsible-content.not-opened:hover {
        width: 100%;
        background-color: rgb(224, 224, 224);
    }

    .collapsible-content-header .number,
    .collapsible-content-header .status,
    .collapsible-content-header .dateTime {
        font-size: 20px;
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
    }
}