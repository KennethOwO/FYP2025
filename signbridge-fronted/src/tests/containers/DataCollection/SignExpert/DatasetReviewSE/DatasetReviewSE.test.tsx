import DatasetReviewSE from "@root/containers/DataCollection/SignExpert/DatasetReviewSE/DatasetReviewSE";
import { create } from "react-test-renderer";
import {
  cleanup,
  render,
  waitFor,
  fireEvent,
  screen,
  act,
} from "@testing-library/react";
import { getAllFormsForSignExpert } from "@root/services/dataset.service";
import "@testing-library/jest-dom";
import CollapsibleForm from "@root/containers/DataCollection/components/CollapsibleForm/CollapsibleForm";

jest.mock("react-i18next", () => ({
  useTranslation: () => {
    const changeLanguage = jest.fn();
    const t = (key: string) => key; // Mock the translation function to return the key
    const i18n = {
      changeLanguage: changeLanguage, // Mock the changeLanguage function
      on: jest.fn(), // Mock the on function
      off: jest.fn(), // Mock the off function
    };

    return { t, i18n };
  },
}));
jest.mock("moment", () => ({
  __esModule: true,
  default: jest.fn(() => ({
    format: jest.fn(() => "2024-05-24 12:00:00"), // Mock the format method
  })),
}));
// Mock the useSpeechRecognition hook
jest.mock("react-speech-recognition", () => ({
  useSpeechRecognition: () => ({
    transcript: "Hello, World!",
    resetTranscript: jest.fn(),
    browserSupportsSpeechRecognition: true,
  }),
}));
jest.mock("@root/services/dataset.service", () => ({
  getAllFormsForSignExpert: jest.fn(),
}));

jest.mock("@root/services/notification.service", () => ({
  CreateNotification: jest.fn(),
  GetUserIdByEmail: jest.fn(() => {
    return {
      data: {
        user_id: 1,
      },
    };
  }),
}));

jest.mock("js-cookie");

describe("Test DataCollectionPublic", () => {
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });
  afterAll(() => {
    // Clean up the DOM
    cleanup();
    // Reset mocks to their initial state
    jest.resetAllMocks();
  });
  it("should render correctly", () => {
    const tree = create(<DatasetReviewSE />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should be able to view submitted dataset collection form", () => {
    const mockForm = {
      form_id: 123,
      submitted_time: new Date(),
      status_SE_en: "Pending",
      status_SE_bm: "Menunggu",
      name: "John Doe",
      email: "john@example.com",
      text_sentence: "Test sentence",
      video_link: "https://example.com/video",
      avatar_link: "https://example.com/avatar",
      user: "signexpert", // or 'admin' based on your requirement
      user_id: 456,
      video_name: "video.mp4",
      avatar_name: "avatar.png",
    };
    (getAllFormsForSignExpert as jest.Mock).mockResolvedValueOnce(mockForm);
    const { getByText, asFragment } = render(<DatasetReviewSE />);
    waitFor(() => {
      // Check if form data is rendered
      expect(getByText(/John Doe/i)).toBeInTheDocument();
      expect(getByText(/Pending/i)).toBeInTheDocument(); // Assuming translation is done correctly
      // Add more assertions as needed
    });
    expect(asFragment).toMatchSnapshot();
  });
  it("should be able to update dataset collection form status", async () => {
    const handleSubmit = jest.fn();
    const props = {
      number: "1",
      form_id: 1,
      dateTime: "2024-05-24T10:30:00Z",
      status_en: "New",
      status_bm: "Baru",
      name: "John Doe",
      email: "john@example.com",
      text: "Lorem ipsum dolor sit amet.",
      video_link: "https://example.com/video",
      avatar_link: "https://example.com/avatar",
      user: "signexpert",
      user_id: 1,
      video_name: "video.mp4",
      avatar_name: "avatar.jpg",
      handleSubmit: handleSubmit,
      handleDelete: jest.fn(),
    };

    render(<CollapsibleForm {...props} />);
    fireEvent.click(screen.getByTestId("collapsible-content_1"));

    await waitFor(() => {
      // Assert that form details are rendered correctly
      expect(screen.getByText(/John Doe/i)).toBeInTheDocument();
      expect(screen.getByText(/New/i)).toBeInTheDocument(); // English status
    });

    // Press the accept button
    fireEvent.click(screen.getByText(/Accept/i)); // Assuming accept button text is "Accept"

    expect(handleSubmit).toHaveBeenCalledTimes(1);
  });

  it("should be able to delete dataset collection form status", async () => {
    const handleSubmit = jest.fn();
    const handleDelete = jest.fn();
    const props = {
      number: "2",
      form_id: 2,
      dateTime: "2024-05-24T10:30:00Z",
      status_en: "New",
      status_bm: "Baru",
      name: "John Doe",
      email: "john@example.com",
      text: "Lorem ipsum dolor sit amet.",
      video_link: "https://example.com/video",
      avatar_link: "https://example.com/avatar",
      user: "signexpert",
      user_id: 1,
      video_name: "video.mp4",
      avatar_name: "avatar.jpg",
      handleSubmit: handleSubmit,
      handleDelete: handleDelete,
    };

    render(<CollapsibleForm {...props} />);
    fireEvent.click(screen.getByTestId("collapsible-content_2"));

    await waitFor(() => {
      // Assert that form details are rendered correctly
      expect(screen.getByText(/John Doe/i)).toBeInTheDocument();
      expect(screen.getByText(/New/i)).toBeInTheDocument(); // English status
    });

    // Press the cancel button
    fireEvent.click(screen.getByText(/Cancel/i)); // Assuming cancel button text is "Cancel"

    expect(handleDelete).toHaveBeenCalledTimes(1);
  });
});
