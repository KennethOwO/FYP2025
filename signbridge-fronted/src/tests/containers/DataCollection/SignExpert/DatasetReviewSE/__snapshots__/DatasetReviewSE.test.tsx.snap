// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DataCollectionPublic should be able to view submitted dataset collection form 1`] = `[Function]`;

exports[`Test DataCollectionPublic should render correctly 1`] = `
<div
  className="dataCollection-bg"
>
  <h1>
    dc_review
  </h1>
  <div
    className="filter-container"
  >
    <div
      className="dataset-filter-group"
    >
      <div
        className="dataset-filter-item"
      >
        <label
          htmlFor="filterFunction"
        />
        <div
          className="ant-select ant-select-outlined css-dev-only-do-not-override-42nv3w ant-select-single ant-select-show-arrow"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          style={
            {
              "height": 40,
              "width": 140,
            }
          }
        >
          <div
            className="ant-select-selector"
            onClick={[Function]}
            onMouseDown={[Function]}
          >
            <span
              className="ant-select-selection-search"
            >
              <input
                aria-autocomplete="list"
                aria-controls="filterFunction_list"
                aria-expanded={false}
                aria-haspopup="listbox"
                aria-owns="filterFunction_list"
                autoComplete="off"
                className="ant-select-selection-search-input"
                disabled={false}
                id="filterFunction"
                onChange={[Function]}
                onCompositionEnd={[Function]}
                onCompositionStart={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onPaste={[Function]}
                readOnly={true}
                role="combobox"
                style={
                  {
                    "opacity": 0,
                  }
                }
                type="search"
                unselectable="on"
                value=""
              />
            </span>
            <span
              className="ant-select-selection-item"
              title="dc_number"
            >
              dc_number
            </span>
          </div>
          <span
            aria-hidden={true}
            className="ant-select-arrow"
            onMouseDown={[Function]}
            style={
              {
                "WebkitUserSelect": "none",
                "userSelect": "none",
              }
            }
            unselectable="on"
          >
            <span
              aria-label="down"
              className="anticon anticon-down ant-select-suffix"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </span>
          </span>
        </div>
      </div>
      <div
        className="dataset-filter-item"
      >
        <div
          className="order-filter"
        >
          <div
            className="ant-select ant-select-outlined css-dev-only-do-not-override-42nv3w ant-select-single ant-select-show-arrow"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            style={
              {
                "height": 40,
                "width": 140,
              }
            }
          >
            <div
              className="ant-select-selector"
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="ant-select-selection-search"
              >
                <input
                  aria-autocomplete="list"
                  aria-controls="undefined_list"
                  aria-expanded={false}
                  aria-haspopup="listbox"
                  aria-owns="undefined_list"
                  autoComplete="off"
                  className="ant-select-selection-search-input"
                  disabled={false}
                  onChange={[Function]}
                  onCompositionEnd={[Function]}
                  onCompositionStart={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  onPaste={[Function]}
                  readOnly={true}
                  role="combobox"
                  style={
                    {
                      "opacity": 0,
                    }
                  }
                  type="search"
                  unselectable="on"
                  value=""
                />
              </span>
              <span
                className="ant-select-selection-item"
                title="ascending"
              >
                ascending
              </span>
            </div>
            <span
              aria-hidden={true}
              className="ant-select-arrow"
              onMouseDown={[Function]}
              style={
                {
                  "WebkitUserSelect": "none",
                  "userSelect": "none",
                }
              }
              unselectable="on"
            >
              <span
                aria-label="down"
                className="anticon anticon-down ant-select-suffix"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiTablePagination-root css-jtlhu6-MuiTablePagination-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
      >
        <div
          className="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
        />
        <p
          className="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
          id=":r3:"
        >
          Rows per page:
        </p>
        <div
          className="MuiInputBase-root MuiInputBase-colorPrimary MuiTablePagination-input css-16c50h-MuiInputBase-root-MuiTablePagination-select"
          onClick={[Function]}
        >
          <div
            aria-controls=":r4:"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby=":r3: :r2:"
            className="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
            id=":r2:"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="combobox"
            tabIndex={0}
          >
            10
          </div>
          <input
            aria-hidden={true}
            aria-invalid={false}
            className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            disabled={false}
            onAnimationStart={[Function]}
            onChange={[Function]}
            tabIndex={-1}
            value={10}
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
        <p
          className="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
        >
          0â€“0 of 0
        </p>
        <div
          className="MuiTablePagination-actions"
        >
          <button
            aria-label="Go to previous page"
            className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            title="Go to previous page"
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowLeftIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
              />
            </svg>
          </button>
          <button
            aria-label="Go to next page"
            className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            title="Go to next page"
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
